src/build_helper/src/git.rs

```diff
@@ -141,28 +162,29 @@ pub fn get_closest_merge_commit(
             // ```
             // Investigate and resolve this issue instead of skipping it like this.
             (channel == "nightly" || !CiEnv::is_rust_lang_managed_ci_job())
-        {
-            git_upstream_merge_base(config, git_dir).unwrap()
-        } else {
-            // For non-CI environments, ignore rust-lang/rust upstream as it usually gets
-            // outdated very quickly.
-            "HEAD".to_string()
+            {
+                git_upstream_merge_base(config, git_dir).unwrap()
```

Comment:
As mentioned in my comment (https://github.com/rust-lang/rust/issues/101907#issuecomment-2692312790, footnote 1) this never executes in practice in (our) CI.

Reply:
Are you sure? We are running that on every pipeline (unless it's beta/stable channel) as far as I know.

Reply:
As Mark explained, `channel` is `"nightly\n"`, so I don't think we are...

Reply:
Ooh, what the hack... lol

Reply:
Maybe that is the reason why sometimes CI fails?